subsys/net/lib/coap/coap.c

```diff
@@ -149,6 +149,20 @@ int coap_packet_init(struct coap_packet *cpkt, uint8_t *data, uint16_t max_len,
 	return 0;
 }
 
+int coap_ack_init(struct coap_packet *cpkt, uint8_t *data, uint16_t max_len,
+		  const struct coap_packet *req, uint8_t ver, uint8_t code)
+{
+	uint16_t id;
+	uint8_t tkl;
+	uint8_t token[COAP_TOKEN_MAX_LEN];
+
+	id = coap_header_get_id(req);
+	tkl = coap_header_get_token(req, token);
```

Comment:
Not sure about this line. The API tends to be generic, yet token inclusion makes it limited to CoAP piggybacked responses. I'd say to include token only if code != 0, otherwise it won't be possible to create an empty ACK.